torchao/float8/float8_tensor.py

```diff
@@ -163,7 +163,8 @@ def forward(
 
         DTensor Invariant: DTensor must always be the outer most tensor subclass
         """
-        tensor_scaled = tensor * scale
+        # scale is float32 thus upcasting tensor to match
+        tensor_scaled = tensor.to(torch.float32) * scale
```

Comment:
without upcasting, the eager numeric is like `-157.00000000000000000000`, compile is like `-157.06507873535156250000`

Reply:
torch.compile upcast tensor ahead, see `tmp0 = tl.load(in_ptr0 + (x0), None).to(tl.float32)` in following output code

```
@triton.jit
def triton_(in_ptr0, in_ptr1, out_ptr0, xnumel, XBLOCK : tl.constexpr):
    xnumel = 24576
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:]
    xmask = tl.full([XBLOCK], True, tl.int1)
    x0 = xindex
    tmp0 = tl.load(in_ptr0 + (x0), None).to(tl.float32)
```